test/errorshow.jl

```diff
@@ -933,3 +933,8 @@ let err_str
     err_str = @except_str "a" + "b" MethodError
     @test occursin("String concatenation is performed with *", err_str)
 end
+
+@testset "issue #47559" begin
+    err = try; invoke(Returns, Tuple{Any,Val{N}} where N, 1, Val(1)) catch ex; ex; end
+    @test startswith(sprint(Base.showerror, err), "MethodError: no method matching Returns(::Any, ::Val{N})")
+end
```

Comment:
Looks like a good start. Unfortunately, I have already worked with Aaron to write a more complete version of this PR (there are some additional subtleties to cleanup too around the use of the parameters). As a minor point of reference, for very new issues, you may want to explicitly confirm on Slack that it is not already being worked on, or at least assign yourself to it, to avoid repeating work.

This test may be helpful though, since we had not written that yet (@apaz-cli).

Reply:
Gotcha - the `errorshow` tests ran through locally, so I thought it should work (the test faillures seem to be unrelated - `ReadOnlyMemoryError` in SparseArrays?). I only looked at the issue in the first place since it had to do with `showerror` of `MethodError`, so I thought it was related to the printing changes.

Assigning myself is not really an option without write access, but I'll ping on slack next time! :sweat_smile: The test more or less follows the pattern at the start of `errorshow.jl`, except with the `sprint` inside of the `@test` to capture that failure correctly.